processor/src/main/java/org/mapstruct/ap/internal/processor/creation/MappingResolverImpl.java

```diff
@@ -122,14 +122,13 @@ public Assignment getTargetAssignment(Method mappingMethod, String mappedElement
         ResolvingAttempt attempt = new ResolvingAttempt(
             sourceModel,
             mappingMethod,
-            mappedElement,
             dateFormat,
             numberFormat,
             sourceRHS,
             criteria
         );
 
-        return attempt.getTargetAssignment( sourceRHS.getSourceType(), targetType );
+        return attempt.getTargetAssignment( sourceRHS.getSourceTypeForMatching(), targetType );
```

Comment:
No problem.. I want to make stuff more compressible. So if I need to add some javadoc or better variable names.. that's the least I can do. Any tip is welcome :)